nodesigner.go

```diff
@@ -44,6 +44,33 @@ func (n *nodeSigner) SignMessage(pubKey *btcec.PublicKey,
 	return sign, nil
 }
 
+// SignCompact signs a double-sha256 digest of the passed msg under the
+// resident node's private key. If the target public key is _not_ the node's
+// private key, then an error will be returned. The returned signature is a
+// pubkey-recoverable signature.
+func (n *nodeSigner) SignCompact(pubKey *btcec.PublicKey,
```

Comment:
As we don't need to adhere to a particular interface (yet) here (which is why the `SignMessage` method is how it is atm), we can omit the public key argument all together. Implicitly, `SignCompact` will always use the resident node's private key.